Frontend/src/Components/Notifications/Notifications.jsx

```diff
@@ -1,67 +1,52 @@
 import React, { useEffect, useState } from "react";
 import { useSocket } from "../Contexts/SocketContext";
 import Loading from "../Loading/Loading.jsx";
+import useLoading from "../CustomHooks/useLoading.jsx";
 import "./Notifications.css";
+import AppHeader from "../Headers/AppHeader";
 
 const API_KEY = import.meta.env.VITE_BACKEND_ADDRESS;
 
 const Notifications = () => {
-  const { socket } = useSocket();
-  const [notifications, setNotifications] = useState([]);
-  const [isLoading, setLoading] = useState(true);
-  const [error, setError] = useState(null);
+  const { notifications, markAsRead } = useSocket();
+  const { isLoading, stopLoading } = useLoading();
 
   useEffect(() => {
-    const fetchNotifications = async () => {
-      setLoading(true);
-      setError(null);
-      try {
-        const response = await fetch(`${API_KEY}/notifications`, {
-          method: "GET",
-          credentials: "include",
-          headers: {
-            Authorization: `Bearer ${localStorage.getItem("token")}`,
-          },
-        });
-        if (!response.ok) {
-          throw new Error("Failed to fetch notifications");
-        }
-        const data = await response.json();
-        setNotifications(data);
-      } catch (err) {
-        setError(err.message);
-      } finally {
-        setLoading(false);
-      }
-    };
+    if (notifications.length > 0) {
+      stopLoading();
+    }
+  }, [notifications]);
 
-    fetchNotifications();
-
-    socket.on("notification", (notification) => {
-      setNotifications((prev) => [notification, ...prev]);
-    });
+  useEffect(() => {
+    const timer = setTimeout(() => {
```

Comment:
maybe something we can think about doing is only marking as read when a notif is clicked on, but clearing the notif count when the notif page is opened. kind of like on workplace!

Reply:
That sounds like a good idea! would definitely implement